spec/formatting.md

```diff
@@ -435,6 +431,18 @@ _Pattern selection_ is not supported for _fallback values_.
 
 ## Pattern Selection
 
+At the start of _pattern selection_,
+if the _message_ contains any _reserved statements_,
+emit an _Unsupported Statement_ error.
+
+If the _message_ being formatted is not _well-formed_ and _valid_,
+the result of pattern selection is a _pattern_ consisting of a single _fallback value_
+using the _message_'s fallback string defined in the _formatting context_
+or if this is not available or empty, the U+FFFD REPLACEMENT CHARACTER `�`.
+
+If the _message_ being formatted does not contain a _matcher_,
+the result of pattern selection is its _pattern_ value.
```

Comment:
I think there might be three cases here (for the sake of clarity)

```suggestion
If the _message_ is a _simple message_,
the result of _pattern selection_ is the _message_ itself.

If the _message_ is a _complex message_
and the _message_ contains no _matcher_,
the result of _pattern selection_ is the _pattern_ value of the _message_.
```

Reply:
I would find it very confusing for pattern selection to result in either a _message_ or a _pattern_, esp. as we also say this: https://github.com/unicode-org/message-format-wg/blob/aa8d29329a66a7e1399cf955a95c0b582cc1b0d5/spec/syntax.md?plain=1#L155-L156

I know that we don't require a data model in the spec, but in places like this I think it's clearer if we don't use language that isn't necessarily related to any data model concepts, such as the difference between a _simple message_ and a _complex message_.

Reply:
I agree, but also want to make life easy on readers of our spec.

Perhaps:

> If the _message_ is a _simple message_, the entire _message_ is the _pattern_.

Reply:
My concerns with your proposal are that
1. As we say in Syntax that the message "contains" a pattern, we should not say here that a message "is" a pattern.
2. By the time we get here, it's almost certain that an implementation is dealing with some data model representation of the pattern, and our own data model does not distinguish between simple and complex messages; both `foo` and `{{foo}}` map to the same construct. Therefore, adding even a reference to _simple message_ is really close to establishing a need for us to add that information to the data model, which we've managed to avoid so far.

Reply:
Fair enough